File: BackendDotNet/DTOs/Response/BloodRequestDto.cs
namespace BackendDotNet.DTOs.Response
{
    public class BloodRequestDto
    {
        public long RequestId { get; set; }
        public HospitalDto Hospital { get; set; } = new();
        public string BloodGroup { get; set; } = string.Empty;
        public int UnitsRequested { get; set; }
        public string Urgency { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public bool IsFulfilled { get; set; }
        public DateTime RequestDate { get; set; }
        public DateTime RequiredDate { get; set; }
        public string HospitalName { get; set; } = string.Empty;
        public string ContactPerson { get; set; } = string.Empty;
        public string ContactNumber { get; set; } = string.Empty;
        public string UrgencyLevel { get; set; } = string.Empty;
    }
}

File: backend spring/src/main/java/com/bloodbank/dto/response/BloodRequestDto.java
package com.bloodbank.dto.response;

import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

// This is the object we will return from our API
@Getter
@Setter
public class BloodRequestDto {
    private Long requestId;
    private HospitalDto hospital; // Using our safe Hospital DTO
    private String bloodGroup;
    private int unitsRequested;
    private String urgency;
    private String status;
    private LocalDateTime createdAt;
    private boolean isFulfilled;
}